torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -496,14 +498,29 @@ std::vector<at::Tensor> FusionKernelRuntime::runWithInput(
       if (iter != runtime_workspace_.tensor_map.end()) {
         fusion_outputs.push_back(iter->second);
       } else {
+        bool empty_type_check = output->getDataType().has_value() &&
+            output->getDataType().value() == DataType::Float;
```

Comment:
Are empty tensors only allowed to have Float type? Can we not have any other type that's empty?

Reply:
This is just legacy code movement but I think we should handle other types in empty tensors once the empty tensor definition is uniform. Will address this in a follow up.